Userland/Libraries/LibJS/Runtime/DataViewPrototype.cpp

```diff
@@ -0,0 +1,286 @@
+        vm.throw_exception<TypeError>(global_object, ErrorType::DetachedArrayBuffer);
+        return {};
+    }
+
+    auto view_offset = view->byte_offset();
+    auto view_size = view->byte_length();
+
+    auto element_size = sizeof(T);
+    if (get_index + element_size > view_size) {
+        vm.throw_exception<RangeError>(global_object, ErrorType::DataViewOutOfRangeByteOffset, get_index, view_size);
+        return {};
+    }
+
+    auto buffer_index = get_index + view_offset;
+    return buffer->get_value<T>(buffer_index, false, ArrayBuffer::Order::Unordered, little_endian);
+}
+
+// 25.3.1.2 SetViewValue ( view, requestIndex, isLittleEndian, type, value ), https://tc39.es/ecma262/#sec-setviewvalue
+template<typename T>
+static Value set_view_value(GlobalObject& global_object, Value request_index, Value is_little_endian, Value value)
+{
+    auto& vm = global_object.vm();
+    auto* view = typed_this(vm, global_object);
+    if (!view)
+        return {};
+
+    auto get_index = request_index.to_index(global_object);
+    if (vm.exception())
+        return {};
+
+    Value number_value;
+    if constexpr (IsIntegral<T> && sizeof(T) == 8)
+        number_value = value.to_bigint(global_object);
+    else
+        number_value = value.to_number(global_object);
+    if (vm.exception())
+        return {};
+
+    auto little_endian = is_little_endian.to_boolean();
+
+    auto buffer = view->viewed_array_buffer();
+    if (buffer->is_detached()) {
+        vm.throw_exception<TypeError>(global_object, ErrorType::DetachedArrayBuffer);
+        return {};
+    }
+
+    auto view_offset = view->byte_offset();
+    auto view_size = view->byte_length();
+
+    auto element_size = sizeof(T);
+    if (get_index + element_size > view_size) {
+        vm.throw_exception<RangeError>(global_object, ErrorType::DataViewOutOfRangeByteOffset, get_index, view_size);
+        return {};
+    }
+
+    auto buffer_index = get_index + view_offset;
+    return buffer->set_value<T>(buffer_index, number_value, false, ArrayBuffer::Order::Unordered, little_endian);
+}
+
+// 25.3.4.5 DataView.prototype.getBigInt64 ( byteOffset [ , littleEndian ] ), https://tc39.es/ecma262/#sec-dataview.prototype.getbigint64
+JS_DEFINE_NATIVE_FUNCTION(DataViewPrototype::get_big_int_64)
+{
+    return get_view_value<i64>(global_object, vm.argument(0), vm.argument(1));
+}
+
+// 25.3.4.6 DataView.prototype.getBigUint64 ( byteOffset [ , littleEndian ] ), https://tc39.es/ecma262/#sec-dataview.prototype.getbiguint64
+JS_DEFINE_NATIVE_FUNCTION(DataViewPrototype::get_big_uint_64)
+{
+    return get_view_value<u64>(global_object, vm.argument(0), vm.argument(1));
+}
+
+// 25.3.4.7 DataView.prototype.getFloat32 ( byteOffset [ , littleEndian ] ), https://tc39.es/ecma262/#sec-dataview.prototype.getfloat32
+JS_DEFINE_NATIVE_FUNCTION(DataViewPrototype::get_float_32)
+{
+    return get_view_value<float>(global_object, vm.argument(0), vm.argument(1).value_or(Value(false)));
```

Comment:
`vm.argument()` already uses `js_undefined()` as OOB fallback, so the `value_or()` will never take effect. Same below.

Reply:
just realised theres no point to doing it manually either, since we use `to_boolean` which converts undefined to false anyways, removed. :)